### PR TITLE
Make `on_slot` return the block with the post header

### PR DESCRIPTION
**Pull request by bkchr**
Before this pr `on_slot` returned the pre block. However this is wrong,
because adding some post digest changes the hash of the header. Thus,
we need to make sure to return the correct block that uses the post
header.

### CODE REVIEW

**Review by octol — APPROVED** [review]

File: primitives/consensus/common/src/block_import.rs [hunk 1]
```diff
@@ -193,16 +193,21 @@ impl<Block: BlockT, Transaction> BlockImportParams<Block, Transaction> {
...
-
-				hdr.hash()
+			self.post_header().hash()
+		}
+	}
+
+	/// Get the post header.
+	pub fn post_header(&self) -> Block::Header {
+		if self.post_digests.is_empty() {
+				self.header.clone()
```
Comment (andresilva) [inline primitives/consensus/common/src/block_import.rs:203]:
```suggestion
			self.header.clone()
```

**Review by andresilva — APPROVED** [review]
Just some wrong indentation, otherwise lgtm.

File: primitives/consensus/common/src/block_import.rs (hunk 1 above)
Comment (bkchr) [reply primitives/consensus/common/src/block_import.rs:203]: EMACS!!!!!

**Comment by bkchr** [comment]
bot merge force

**Pull request merged**